superset/utils/core.py

```diff
@@ -617,7 +618,14 @@ def handle_timeout(  # pylint: disable=unused-argument
         self, signum: int, frame: Any
     ) -> None:
         logger.error("Process timed out")
-        raise SupersetTimeoutException(self.error_message)
+        raise SupersetTimeoutException(
+            SupersetError(
+                error_type=SupersetErrorType.BACKEND_TIMEOUT_ERROR,
+                message=self.error_message,
+                level=ErrorLevel.ERROR,
+                extra={"timeout": self.seconds,},
+            )
```

Comment:
Using two classes to initialize things is kind of weird to me. Can you pass the extras as arguments to `SupersetTimeoutException` and let `SupersetTimeoutException` to enforce the structure for `SupersetError`?

Reply:
done